src/main/java/com/code_intelligence/jazzer/junit/FuzzerDictionary.java

```diff
@@ -0,0 +1,159 @@
+ */
+public class FuzzerDictionary {
+  private static final String DICTIONARY_PREFIX = "jazzer-";
+  private static final String DICTIONARY_SUFFIX = ".dict";
+
+  /**
+   * Create a temporary dictionary file for use during a fuzzing run based on the tokens found
+   * within {@code inline} and {@code files}.
+   *
+   * @param inline List of {@link WithDictionary} annotations that directly hold static token values
+   *     to use in the dictionary
+   * @param files List of {@link WithDictionaryFile} annotations that reference dictionary files to
+   *     include
+   * @return Optional containing the path to the created file, or nothing if {@code inline} and
+   *     {@code files} are both empty
+   * @throws IOException
+   */
+  public static Optional<String> createDictionaryFile(
+      List<WithDictionary> inline, List<WithDictionaryFile> files) throws IOException {
+    int sources = inline.size() + files.size();
+    if (sources == 0) {
+      return Optional.empty();
+    }
+
+    Stream<String> joined = Stream.concat(getInlineTokens(inline), getFileTokens(files));
+
+    File f = File.createTempFile(DICTIONARY_PREFIX, DICTIONARY_SUFFIX);
+    f.deleteOnExit();
+    Log.info(String.format("Creating merged dictionary from %d sources", sources));
+
+    try (OutputStream out = Files.newOutputStream(f.toPath())) {
+      joined.forEach(
+          (token) -> {
+            try {
+              // the tokens will come in without newlines attached, so we append them here before
+              // writing
+              String line = token.concat("\n");
+              out.write(line.getBytes());
+            } catch (IOException e) {
+              throw new UncheckedIOException(e);
+            }
+          });
+    }
+    return Optional.of(f.getPath());
+  }
+
+  /**
+   * Gets the inlined arrays from each annotation, flattens them into a single stream, and wraps the
+   * elements in double quotes to comply with libfuzzer's dictionary syntax
+   *
+   * @param inline List of {@link WithDictionary} annotations to extract from
+   * @return stream of all the tokens from each of the elements of {@code inline}
+   */
+  private static Stream<String> getInlineTokens(List<WithDictionary> inline) {
+    return inline.stream()
+        .map(WithDictionary::tokens)
+        .flatMap(Arrays::stream)
+        .map((token) -> String.format("\"%s\"", token));
+  }
+
+  /**
+   * Gets the individual lines from each of the specified dictionary files
+   *
+   * @param files List of {@link WithDictionaryFile} annotations indicating which files to use
+   * @return stream of all lines from each of the files
+   */
+  private static Stream<String> getFileTokens(List<WithDictionaryFile> files) {
+    return files.stream()
+        .map(WithDictionaryFile::resourcePath)
+        .map(FuzzerDictionary::tokensFromFile)
+        .flatMap(List::stream);
+  }
+
+  private static List<String> tokensFromFile(String path) {
```

Comment:
This makes it more clear that this method operates on resources:
```suggestion
  private static List<String> tokensFromResource(String absoluteResourcePath) {
```